src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -174,10 +176,22 @@ public LinkedPointSet(LinkedPointSet sourceLinkage, WebMercatorGridPointSet subG
         distances0_mm = new int[nCells];
         distances1_mm = new int[nCells];
 
+        // mark all points as unlinked, in case part of the subGrid lies outside the super grid. Points within the super
+        // grid will be overwritten below
+        Arrays.fill(edges, -1);
```

Comment:
Maybe we should be using a symbolic constant like UNLINKED for this, but I guess we have hard-coded -1s all over.